### PR TITLE
improve buildscript

### PR DESCRIPTION
**Pull request by vchuravy**
This cleans the buildscript a bit up.

@pluskid This should definitely run through the CI first. I tested it locally against prebuild versions of Julia, but I am not a 100% sure that `MakeTargets` works as I expect.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -44,48 +44,69 @@ if !libmxnet_detected
   end
 
   if blas_vendor == :unknown
-    error("Julia is build with an unkown blas library ($blas_path).\n Automatic building of libmxnet is not yet supported.")
-  elseif blas_vendor != :openblas64 && blas_vendor != :openblas
-    warn("Unsure if we can build against $blas_vendor.")
+    info("Julia is build with an unkown blas library ($blas_path).")
```
Comment (tkelman) [inline deps/build.jl]: built?

File: deps/build.jl [hunk 2]
```diff
@@ -44,48 +44,69 @@ if !libmxnet_detected
...
-          `cp lib/libmxnet.so $_libdir`
+          if USE_JULIA_BLAS
+          MakeTargets("USE_BLAS=$blas_name -j$(nprocs())", env=Dict(
+                      "MSHADOW_LDFLAGS" => blas_path,
+                      "MSHADOW_CFLAGS" => ilp64,
+                     ))
+          else
+            `make -j$(nprocs())`
+          end
+          `rm $_libdir/libmxnet.so`
```
Comment (pluskid) [inline deps/build.jl]: `rm -f`, otherwise the command will fail if the file does not exist.

**Comment by pluskid** [comment]
merged locally and tested after fixing the comments above.

**Comment by pluskid** [comment]
@vchuravy It seems Travis CI still don't like it. It says symbol `cblas_sgemm` not found. Maybe the openblas library is not properly linked.

**Comment by tkelman** [comment]
The generic linux binary renames all symbols to avoid colliding with non ILP64 libraries also being loaded

**Pull request merged**